.travis.yml

```diff
@@ -1,58 +1,44 @@
-    - DOCKER_PACKAGES="build-essential cmake git libcap-dev bsdmainutils ninja-build curl git ca-certificates ccache libuv1-dev"
-  matrix:
-      #- HOST=x86_64-w64-mingw32
-      - HOST=x86_64-unknown-linux-gnu STATIC_LINK=OFF
-      #- HOST=x86_64-unknown-linux-gnu STATIC_LINK=ON
-      #- HOST=x86_64-unknown-linux-gnu STATIC_LINK=OFF NINJA=ninja
-      #- HOST=x86_64-unknown-linux-gnu STATIC_LINK=ON NINJA=ninja
-      #- HOST=x86_64-unknown-linux-gnu STATIC_LINK=OFF NETNS=ON
-      #- HOST=x86_64-unknown-linux-gnu STATIC_LINK=ON NETNS=ON
-      #- HOST=x86_64-unknown-linux-gnu STATIC_LINK=OFF NINJA=ninja NETNS=ON
-      #- HOST=x86_64-unknown-linux-gnu STATIC_LINK=ON NINJA=ninja NETNS=ON
-      #- HOST=x86_64-unknown-linux-gnu STATIC_LINK=OFF SHARED_LIB=ON
-      #- HOST=x86_64-unknown-linux-gnu STATIC_LINK=ON SHARED_LIB=ON
-      #- HOST=x86_64-unknown-linux-gnu STATIC_LINK=OFF NINJA=ninja SHARED_LIB=ON
-      #- HOST=x86_64-unknown-linux-gnu STATIC_LINK=ON NINJA=ninja SHARED_LIB=ON
-      #- HOST=x86_64-unknown-linux-gnu STATIC_LINK=OFF NETNS=ON SHARED_LIB=ON
-      #- HOST=x86_64-unknown-linux-gnu STATIC_LINK=ON NETNS=ON SHARED_LIB=ON
-      #- HOST=x86_64-unknown-linux-gnu STATIC_LINK=OFF NINJA=ninja NETNS=ON SHARED_LIB=ON
-      #- HOST=x86_64-unknown-linux-gnu STATIC_LINK=ON NINJA=ninja NETNS=ON SHARED_LIB=ON
-      #- HOST=x86_64-apple-darwin11 OSX_SDK=10.12
-install:
-    - env | grep -E '^(CCACHE_|DISPLAY|CONFIG_SHELL)' | tee /tmp/env
-    - if [[ $HOST = *-mingw32 ]]; then DOCKER_ADMIN="--cap-add SYS_ADMIN"; fi
-    - DOCKER_ID=$(docker run $DOCKER_ADMIN -idt --mount type=bind,src=$TRAVIS_BUILD_DIR,dst=$TRAVIS_BUILD_DIR --mount type=bind,src=$CCACHE_DIR,dst=$CCACHE_DIR -w $TRAVIS_BUILD_DIR --env-file /tmp/env ubuntu:18.04)
-    - DOCKER_EXEC="docker exec $DOCKER_ID"
-    - if [ -n "$DPKG_ADD_ARCH" ]; then $DOCKER_EXEC dpkg --add-architecture "$DPKG_ADD_ARCH" ; fi
-    - travis_retry $DOCKER_EXEC apt-get update
-    - travis_retry $DOCKER_EXEC apt-get install --no-install-recommends --no-upgrade software-properties-common -qq $PACKAGES $DOCKER_PACKAGES
-#    - travis_retry $DOCKER_EXEC add-apt-repository ppa:ubuntu-toolchain-r/test
-#    - travis_retry $DOCKER_EXEC apt-get update
-#    - travis_retry $DOCKER_EXEC apt-get install --no-install-recommends --no-upgrade -qq gcc-6 g++-6
+  - STATIC_LINK=OFF
+  - STATIC_LINK=OFF
+  - STATIC_LINK=ON
+  - STATIC_LINK=OFF NINJA=ninja
+  - STATIC_LINK=ON NINJA=ninja
+  - STATIC_LINK=OFF NETNS=ON
+  - STATIC_LINK=ON NETNS=ON
+  - STATIC_LINK=OFF NINJA=ninja NETNS=ON
+  - STATIC_LINK=ON NINJA=ninja NETNS=ON
+  - STATIC_LINK=OFF SHARED_LIB=ON
+  - STATIC_LINK=ON SHARED_LIB=ON
+  - STATIC_LINK=OFF NINJA=ninja SHARED_LIB=ON
+  - STATIC_LINK=ON NINJA=ninja SHARED_LIB=ON
+  - STATIC_LINK=OFF NETNS=ON SHARED_LIB=ON
+  - STATIC_LINK=ON NETNS=ON SHARED_LIB=ON
+  - STATIC_LINK=OFF NINJA=ninja NETNS=ON SHARED_LIB=ON
+  - STATIC_LINK=ON NINJA=ninja NETNS=ON SHARED_LIB=ON
```

Comment:
do these combinations really make sense? we probably don’t need to test with/without ninja etc

Reply:
Probably right, I don't think it's support breaks too often (I'm not 100% it ever has but memory think either it did or something adjacent to it did) but would be nice to know when it does (if that's possible)